apps/src/codebridge/ControlButtons/index.tsx

```diff
@@ -27,6 +27,13 @@ const ControlButtons: React.FunctionComponent = () => {
   const hasNextLevel = useAppSelector(
     state => nextLevelId(state) !== undefined
   );
+  const hasPredictResponse = useAppSelector(
+    state => !!state.predictLevel.response
+  );
+  const isPredictLevel = useAppSelector(
+    state => state.lab.levelProperties?.predictSettings?.isPredictLevel
+  );
+  const disableRunAndTest = loading || (isPredictLevel && !hasPredictResponse);
```

Comment:
Can you explain why we don't need to consider whether `predictAnswerIsLocked` here?

Reply:
The answer can't be locked if we don't have a predict response.

Reply:
Ah, right. Okay. This disables the "run and test" button on predict levels _before_ you've submitted a response. But you can continue running and testing once you've submitted a response (even if it's un-editable). Thanks!